shared/src/main/java/org/tiqr/service/authentication/AuthenticationService.java

```diff
@@ -233,17 +231,17 @@ protected AuthenticationError doInBackground(Void... voids) {
                     byte[] postData = Utils.keyValueMapToByteArray(nameValuePairs);
 
                     URL authenticationURL = new URL(challenge.getIdentityProvider().getAuthenticationURL());
-                    HttpsURLConnection httpsURLConnection = (HttpsURLConnection)authenticationURL.openConnection();
-                    httpsURLConnection.setRequestMethod("POST");
-                    httpsURLConnection.setRequestProperty("ACCEPT", "application/json");
-                    httpsURLConnection.setRequestProperty("X-TIQR-Protocol-Version", Constants.PROTOCOL_VERSION);
-                    httpsURLConnection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
-                    httpsURLConnection.setRequestProperty("Content-Length", String.valueOf(postData.length));
-                    httpsURLConnection.setDoOutput(true);
-                    httpsURLConnection.getOutputStream().write(postData);
-
-                    String response = Utils.urlConnectionResponseAsString(httpsURLConnection);
-                    String versionHeader = httpsURLConnection.getHeaderField("X-TIQR-Protocol-Version");
+                    HttpURLConnection httpURLConnection = (HttpURLConnection)authenticationURL.openConnection();
+                    httpURLConnection.setRequestMethod("POST");
+                    httpURLConnection.setRequestProperty("ACCEPT", "application/json");
+                    httpURLConnection.setRequestProperty("X-TIQR-Protocol-Version", Constants.PROTOCOL_VERSION);
+                    httpURLConnection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+                    httpURLConnection.setRequestProperty("Content-Length", String.valueOf(postData.length));
+                    httpURLConnection.setDoOutput(true);
```

Comment:
You are not supposed to write to a GET request, so I guess that was the problem :)